### PR TITLE
Ensure that non-matmul schedulers reject matmul fusions

### PR DESCRIPTION
**Pull request by jacobhinkle**
This ensures that if MmaOp or MatmulOp occur in the fusion, only ExprEval or Matmul schedulers will consider it.

### CODE REVIEW

**Comment by jacobhinkle** [comment]
Included test fails on main

File: tests/cpp/test_matmul_scheduler.cpp [hunk 1]
```diff
@@ -2815,6 +2815,51 @@ TEST_F(MatmulSchedulerPluginTest, BasicMatmul) {
       executor_cache.fusion(), outputs, {t0, t1}, {tref}, __LINE__, __FILE__);
 }
 
+// Test that we can segment a fusion that has a MatmulOp with epilogue
+// TODO: Once we can control the ExprEval and Matmul schedulers via options, run
+// this test with all three combinations (with and without each scheduler, but
+// at least one enabled).
```
Comment (jacobhinkle) [inline tests/cpp/test_matmul_scheduler.cpp:2821]: A version of this test that works like this TODO suggests is here (passing on that branch): https://github.com/NVIDIA/Fuser/pull/2236/files#diff-79605590906bd352b3f5d8798d0f16a00350ab8e740cdc0e64a89a7f8b34d204R341-R351

File: tests/cpp/test_matmul_scheduler.cpp [hunk 2]
```diff
@@ -2815,6 +2815,51 @@ TEST_F(MatmulSchedulerPluginTest, BasicMatmul) {
...
+  // rejects, then Pointwise must not accept this unsegmented fusion.
+  tv1 = castOp(DataType::Half, sin(tv1));
+
+  auto tv2 = matmul(tv0, tv1);
+
+  fusion->addOutput(tv2);
+
+  NVF_CHECK(
+      1 == ir_utils::getOpsOfType<MatmulOp>(fusion.get()).size(),
+      "matmul fusion must have at least one MmaOp");
```
Comment (Priya2698) [inline tests/cpp/test_matmul_scheduler.cpp:2843]: nit: Easier to read with the value as rhs
```Suggestion
  NVF_CHECK(
      ir_utils::getOpsOfType<MatmulOp>(fusion.get()).size() == 1,
      "matmul fusion must have at least one matmul op.");
```

File: tests/cpp/test_matmul_scheduler.cpp [hunk 3]
```diff
@@ -2815,6 +2815,51 @@ TEST_F(MatmulSchedulerPluginTest, BasicMatmul) {
...
+  auto t0 = at::randn({M, K}, options);
+  auto t1 = at::randn({K, N}, options);
+  auto tref = at::matmul(t0, sin(t1));
+
+  auto outputs = executor_cache.runFusionWithInputs({t0, t1});
+
+  // TODO: check vectorization if fusion is enabled
+  // checkUnsegmentedVectorization(executor_cache, 8, 8, 8);
+
+  NVF_CHECK(outputs[0].allclose(tref, 0.001, 0.001));
```
Comment (Priya2698) [inline tests/cpp/test_matmul_scheduler.cpp:2860]: We can use `testValidate` here, to avoid setting tolerance values.

File: csrc/ir/utils.cpp [hunk 4]
```diff
@@ -456,6 +456,10 @@ bool hasAnyReductionOps(Fusion* fusion) {
   return hasOpsOfType<ReductionOp, GroupedReductionOp, WelfordOp>(fusion);
 }
 
+bool hasAnyMatmulOps(Fusion* fusion) {
+  return hasOpsOfType<MmaOp, MatmulOp>(fusion);
```
Comment (Priya2698) [inline csrc/ir/utils.cpp:460]: We can add `LinearOp` here now.

**Review by Priya2698 — CHANGES_REQUESTED** [review]

**Review by Priya2698 — APPROVED** [review]
LGTM.

**Comment by jacobhinkle** [comment]
!build

**Pull request merged**